airflow/models/dagbag.py

```diff
@@ -95,6 +95,7 @@ def __init__(
         read_dags_from_db: bool = False,
         store_serialized_dags: Optional[bool] = None,
         load_op_links: bool = True,
+        skip_check_name: bool = conf.getboolean('core', 'DAG_DISCOVERY_SKIP_CHECK_NAME'),
```

Comment:
We shouldn't be adding a new flag for this -- there is alread a setting called "safemode".